config/crd-overlay.yml

```diff
@@ -18,3 +18,11 @@ spec:
 #@overlay/remove
 #@overlay/match missing_ok=True
 status:
+
+#@overlay/match by=overlay.subset({"metadata":{"name":"internalpackagemetadata.internal.packaging.carvel.dev"}}), expects="0+"
```

Comment:
Do we wanna do expects=1 here so that we blow up if we ever match multiple CRDs?

Reply:
We don't generate the crds as part of ./hack/deploy-test.sh, so it would fail since the generated crds are produced.

Reply:
Thoughts on generating them as part of deploy-test? I go back and forth here since it would be a good way to ensure crds are regenerated when changes are made and to ensure you're testing with the same deployment that would be included in a release, but I can also see arguments against generating them there.

Reply:
I went ahead and added a step to check if generation has happened. This means that the overlay will be used strictly for purposes of generating the crds and not as part of the actual deploy process. The pro for this would be having more control with expects=1+ for the internalpackagemetadata specific overlay. The con is knowing we need to exclude it for all the places ytt uses it. Alternatively, we could add the overlay to a separate directory to keep it separate from config folder.